src-tauri/tests/tests/credential_offer.rs

```diff
@@ -0,0 +1,357 @@
+                                                "locale": "en-US"
+                                            }
+                                        ]
+                                    },
+                                    "family_name": {
+                                        "display": [
+                                            {
+                                                "name": "Surname",
+                                                "locale": "en-US"
+                                            }
+                                        ]
+                                    },
+                                    "degree": {},
+                                    "gpa": {
+                                        "display": [
+                                            {
+                                                "name": "GPA"
+                                            }
+                                        ]
+                                    }
+                                })),
+                            },
+                            None,
+                        )
+                            .into(),
+                    }),
+                    scope: Some("UniversityDegreeCredential".to_string()),
+                    cryptographic_binding_methods_supported: Some(vec!["did".to_string()]),
+                    cryptographic_suites_supported: Some(vec!["ES256K".to_string()]),
+                    proof_types_supported: None,
+                    display: None,
+                }],
+                credential_issuer: "https://server.example.com".parse().unwrap(),
+                authorization_server: None,
+                batch_credential_endpoint: None,
+                deferred_credential_endpoint: None,
+                display: None,
+            }),
+        )
+        .expect(2)
+        .mount(&mock_server)
+        .await;
+
+    // TODO: assert that function download_asset() is never called (through spy?)
+
+    let mut state = AppState {
+        managers: test_managers(vec![]),
+        ..AppState::default()
+    };
+
+    assert!(read_credential_offer(
+        state.borrow_mut(),
+        Action {
+            r#type: ActionType::ReadCredentialOffer,
+            payload: Some(json!({"credential_offer_uri": format!("{}/offer/1", &mock_server.uri())})),
+        },
+    )
+    .await
+    .is_ok());
+}
+
+#[tokio::test]
+async fn move_downloaded_logo_asset_out_of_tmp_folder() {
+    *ASSETS_DIR.lock().unwrap() = TempDir::new().unwrap().into_path();
+
+    let mut state = AppState {
+        managers: test_managers(vec![]),
+        ..AppState::default()
+    };
+
+    assert!(send_credential_request(
+        state.borrow_mut(),
+        Action {
+            r#type: ActionType::CredentialOffersSelected,
+            payload: Some(json!({"offer_indices": [0]})),
+        },
+    )
+    .await
+    .is_ok());
+}
```

Comment:
This test is failing because `send_credential_request` is expecting there to be a `CurrentUserPrompt::CredentialOffer` in the `AppState`. It uses the `offer_indices` it receives to find de credentials in the offer (possible misuses of `CurrentUserPrompt` but that's how it works.

This could be solved in the test by using this `AppState` instead:
```
    let mut state = AppState {
        managers: test_managers(vec![]),
        current_user_prompt: Some(CurrentUserPrompt::CredentialOffer {
            issuer_name: mock_server.uri().to_string(),
            logo_uri: Some("https://example.org/logo_uri".to_string()),
            credential_offer: CredentialOffer::<CredentialFormats> {
                credential_issuer: "https://example.org/credential_issuer".parse().unwrap(),
                credentials: vec![CredentialsObject::ByReference("UniversityDegreeCredential".to_string())],
                grants: None,
            },
        }),
        ..AppState::default()
    };
```

Obviously the `MockServer` needs to be able to respond to the respective calls that `send_credential_request` will make.